### ISSUE TITLE
ib device rex0 not found even though i load the rdma_rxe , and the sh:echo: I/O  error

### ISSUE THREAD

**Issue by btlcmr0702** [SoftRoCE/rxe-dev#49]
hi,i followed the step to finish configuration of SoftRoCE , but i met some problems i can't fix.
1. the problem abt I/O error?
btl@ubuntu:~$ sudo rxe_cfg start
sh: echo: I/O error
sh: echo: I/O error
sh: echo: I/O error
sh: echo: I/O error
Use of uninitialized value $rmtu in string at /bin/rxe_cfg line 371.
  Name  Link  Driver  Speed  NMTU  IPv4_addr      RDEV  RMTU  
  eth0  yes   e1000          1500  114.212.22.35  rxe0  

2.i can't add device because the I/O error
btl@ubuntu:~$ sudo rxe_cfg add enp5s0
sh: echo: I/O error

3.i have rxe0 device, but it says IB device rxe0 not found
btl@ubuntu:~$ sudo rxe_cfg status
Use of uninitialized value $rmtu in string at /bin/rxe_cfg line 371.
  Name  Link  Driver  Speed  NMTU  IPv4_addr      RDEV  RMTU  
  eth0  yes   e1000          1500  114.212.22.35  rxe0  
btl@ubuntu:~$ ibv_devices
libibverbs: Warning: couldn't load driver 'rxe': librxe-rdmav2.so: cannot open shared object file: No such file or directory
libibverbs: Warning: no userspace device-specific driver found for /sys/class/infiniband_verbs/uverbs0
    device                 node GUID
    ------              ----------------

---

here are some info abt my configuration
btl@ubuntu:~$ ls /sys/class/infiniband
rxe0
btl@ubuntu:~$ lsmod | grep rxe
rdma_rxe              102400  0 
ip6_udp_tunnel         16384  1 rdma_rxe
udp_tunnel             16384  1 rdma_rxe
ib_core               208896  6 rdma_cm,ib_cm,iw_cm,ib_uverbs,rdma_rxe,rdma_ucm
btl@ubuntu:~$ uname -r
4.7.0-rc3+

---

i will wait for some replies, thank u ~

**Comment by monis410** [comment]
please uninstall and than install librxe-dev and let me know what happens
Make sure you have the latest code (head should be 227e3c49b6e423c066e1e1887fe30c8261f63cbd)

**Comment by btlcmr0702** [comment]
hi,thank u for the replies
i check out the version and it is the latest code
btl@ubuntu:~/librxe-dev$ git log
commit 227e3c49b6e423c066e1e1887fe30c8261f63cbd

and then  i delete the old librxe-dev , install a new one by git clone https://github.com/SoftRoCE/librxe-dev.git  but it seems doesn't work, the problem will happen just like before

**Comment by btlcmr0702** [comment]
hi,thank u for ur replies
i check out the version and it is the latest code
btl@ubuntu:~/librxe-dev$ git log
commit 227e3c49b6e423c066e1e1887fe30c8261f63cbd

and then i delete the old librxe-dev , install a new one by git clone https://github.com/SoftRoCE/librxe-dev.git but it seems doesn't work, the problem will happen just like before



________________________________
发件人: Moni Soua <notifications@github.com>
发送时间: 2016年7月19日 7:47
收件人: SoftRoCE/rxe-dev
抄送: btlcmr0702; Author
主题: Re: [SoftRoCE/rxe-dev] ib device rex0 not found even though i load the rdma_rxe , and the sh:echo: I/O error (#49)


please uninstall and than install librxe-dev and let me know what happens
Make sure you have the latest code (head should be 227e3c49b6e423c066e1e1887fe30c8261f63cbd)

―

**Comment by monis410** [comment]
Hi
any chance that we schedule an online session?
we can do it with google hangout

thanks

**Comment by btlcmr0702** [comment]
wow,really thank u :) i didn't think this problem would be so troublesome that need a session.
if it is convenient to u,maybe we can have a session tomorrow and i will inform u in advance.

**Comment by monis410** [comment]
sure, please coordinate via email monis@mellanox.com

**Comment by btlcmr0702** [comment]
ok, i have send a request on hangout , if u r free u could accept it and we can have a session

**Comment by monis410** [comment]
I received the invitation but it is unfortunately in a language I don't speak (Chinese?).
Can you please send me an email from your private mail account to which I can reply?

**Comment by mcfatealan** [comment]
Met the same problem today. I tried to copy /usr/lib64/librxe-rdmav2.so to /usr/lib/librxe-rdmav2.so, and the problem seemed to be solved?

**Comment by sbates130272** [comment]
Hi 

I am confirming a similar issue. The librxe\* files are getting installed in /usr/lib64 and I think they need to be installed in /usr/lib on certain systems. If I copy or symlink the files over then things work great for me!

Stephen

**Comment by mcfatealan** [comment]
@sbates130272 seems still having problems even walking around this issue?

**Comment by sbates130272** [comment]
No. If I copy or symlink the files over then everything works well.
